### PR TITLE
Add category model

### PR DESCRIPTION
**Pull request by padu12**

### CODE REVIEW

File: src/main/java/com/kaziamyr/onlinebookstore/repository/CategoryRepository.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.kaziamyr.onlinebookstore.repository;
+
+import com.kaziamyr.onlinebookstore.model.Category;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface CategoryRepository extends JpaRepository<Category, Long> {
+    Category getCategoryById(Long id);
```
Comment (boroda4436) [inline src/main/java/com/kaziamyr/onlinebookstore/repository/CategoryRepository.java]: we have default method `getReferenceById`

File: src/main/java/com/kaziamyr/onlinebookstore/controller/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "api/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping
+    @Operation(summary = "Create category. Validation is present")
+    public CategoryDto createCategory(@RequestBody @Valid SaveCategoryRequestDto request) {
```
Comment (andrii-hoienko) [inline src/main/java/com/kaziamyr/onlinebookstore/controller/CategoryController.java]:
```suggestion
    public CategoryDto create(@RequestBody @Valid SaveCategoryRequestDto request) {
```

Comment (andrii-hoienko) [inline src/main/java/com/kaziamyr/onlinebookstore/controller/CategoryController.java]: It is clear that you work with category here, not necessary to add it

File: src/main/java/com/kaziamyr/onlinebookstore/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -19,12 +22,18 @@
...
     private final BookRepository bookRepository;
+    private final CategoryRepository categoryRepository;
     private final BookMapper bookMapper;
     private final SpecificationProvider<Book> specificationProvider;
 
     @Override
     public BookDto save(CreateBookRequestDto requestDto) {
         Book book = bookMapper.toModel(requestDto);
+        book.setCategories(
+                Stream.of(requestDto.getCategoryIds())
```
Comment (andrii-hoienko) [inline src/main/java/com/kaziamyr/onlinebookstore/service/impl/BookServiceImpl.java]: move to separate method

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, some minor improvements required

**Review by andrii-hoienko — APPROVED** [review]

**Pull request merged**